app/Traits/Imageable.php

```diff
@@ -20,115 +20,18 @@
 namespace App\Traits;
 
 use App\Libraries\ImageProcessor;
-use App\Libraries\StorageWithUrl;
 
 trait Imageable
 {
-    protected $_storage = null;
+    use Uploadable {
+        storeFile as _storeFile; // rename storeFile in the Uploadable trait so we can override it
+    }
 
     /**
      * Returns maximum dimensions of the image as an array of [width, height].
      */
     abstract public function getMaxDimensions();
 
-    /**
-     * Returns maximum size of the file in bytes. Defaults to 1 MB.
-     */
-    public function getMaxFileSize()
-    {
-        return 1000000;
-    }
-
-    /**
-     * Returns root path of where the files are to be stored.
-     */
-    abstract public function getFileRoot();
-
-    public function getFileId()
-    {
-        return $this->id;
-    }
-
-    /**
-     * Returns a hash with contents of at least 'hash' and 'ext' if there's
-     * image or otherwise null.
-     *
-     * Assumes attributes 'hash' and 'ext' of the object by default.
-     */
-    public function getFileProperties()
-    {
-        if (present($this->hash) === false || present($this->ext) === false) {
-            return;
-        }
-
-        return [
-            'hash' => $this->hash,
-            'ext' => $this->ext,
-        ];
-    }
-
-    /**
-     * Sets file properties. Either a hash of 'hash' and 'ext' or null.
-     *
-     * Assumes attributes 'hash' and 'ext' of the object by default.
-     */
-    public function setFileProperties($props)
-    {
-        $this->hash = $props['hash'] ?? null;
-        $this->ext = $props['ext'] ?? null;
-    }
-
-    public function storage()
-    {
-        if ($this->_storage === null) {
-            $this->_storage = new StorageWithUrl();
-        }
-
-        return $this->_storage;
-    }
-
-    public function fileDir()
-    {
-        return $this->getFileRoot().'/'.$this->getFileId();
-    }
-
-    public function fileName()
-    {
-        return $this->getFileProperties()['hash'].'.'.$this->getFileProperties()['ext'];
-    }
-
-    public function filePath()
-    {
-        return $this->fileDir().'/'.$this->fileName();
-    }
-
-    public function fileUrl()
-    {
-        if ($this->getFileProperties() === null) {
-            return;
-        }
-
-        return $this->storage()->url($this->filePath());
-    }
-
-    public function deleteWithFile()
-    {
-        $this->deleteFile();
-
-        return $this->delete();
-    }
-
-    public function deleteFile()
-    {
-        if ($this->getFileProperties() === null) {
-            return;
-        }
-
-        $this->setFileProperties(null);
-
-        return $this->storage()->deleteDirectory($this->fileDir());
-    }
-
     public function storeFile($filePath)
     {
         $image = new ImageProcessor($filePath, $this->getMaxDimensions(), $this->getMaxFileSize());
```

Comment:
maybe just call `return $this->_storeFile($image->inputPath);` after the process call instead of duplicating the whole thing?